src/satosa/frontends/saml2.py

```diff
@@ -299,12 +300,28 @@ def _handle_authn_response(self, context, internal_response, idp):
             attributes_to_remove = custom_release.get("exclude", [])
             for k in attributes_to_remove:
                 ava.pop(k, None)
-
-        name_id = NameID(text=internal_response.user_id,
-                         format=hash_type_to_saml_name_id_format(
-                             internal_response.user_id_hash_type),
-                         sp_name_qualifier=None,
-                         name_qualifier=None)
+        try:
+            config_nameidformat = self.config['idp_config']['service']['idp']['name_id_format'][0]
```

Comment:
`self.idp_config` is already set, this could be 

```python
config_nameidformat = self.idp_config['service']['idp']['name_id_format'][0]
```

However, after setting `name_id` we retrieve that path again. So maybe this should be

```python
idp_conf = self.idp_config.get('service', {}).get('idp', {})
config_nameidformat = idp_conf.get('name_id_format')
if isinstance(config_nameidformat, list):
    config_nameidformat = config_nameidformat[0]
satosa_logging(logger, logging.DEBUG, "Frontend IDP nameid format: {}".format(config_nameidformat), context.state)
# ...
policies = idp_conf.get('policy', {})
```